File: Serenity.Script.UI/Editor/DecimalEditor.cs
using jQueryApi;
using System;
using System.ComponentModel;
using System.Runtime.CompilerServices;

namespace Serenity
{
    [Imported(ObeysTypeSystem = true)]
    [Editor, DisplayName("Ondalıklı Sayı"), OptionsType(typeof(DecimalEditorOptions))]
    [Element("<input type=\"text\"/>")]
    public class DecimalEditor : Widget<DecimalEditorOptions>, IDoubleValue
    {
        static DecimalEditor()
        {
            Q.Prop(typeof(DecimalEditor), "value");
        }

        public DecimalEditor(jQueryObject input, DecimalEditorOptions opt)
            : base(input, opt)
        {
            input.AddClass("decimalQ");

            dynamic numericOptions = jQuery.ExtendObject(DefaultAutoNumericOptions(), new
            {
                vMin = options.MinValue,
                vMax = options.MaxValue
            });

            if (options.Decimals.HasValue)
                numericOptions.mDec = options.Decimals.Value;

            if (options.PadDecimals.HasValue)
                numericOptions.aPad = options.PadDecimals.Value;

            ((dynamic)input).autoNumeric(numericOptions);
        }

        public Double? Value
        {
            get
            {
                var val = ((dynamic)element).autoNumeric("get");
                if (val == null || val.Length == 0)
                    return null;
                else
                    return Double.Parse(val);
            }
            set
            {
                if (value == null || value.As<string>() == "")
                    element.Value("");
                else
                    ((dynamic)element).autoNumeric("set", value);
            }
        }

        public bool IsValid
        {
            get
            {
                return !Double.IsNaN(Value.As<double>());
            }
        }

        public static dynamic DefaultAutoNumericOptions()
        {
            return new
            {
                aDec = Q.Culture.DecimalSeparator,
                altDec = Q.Culture.DecimalSeparator == "." ? "," : ".",
                aSep = Q.Culture.DecimalSeparator == "." ? "," : ".",
                aPad = true
            };
        }
    }

    [Imported, Serializable]
    public class DecimalEditorOptions
    {
        public string MinValue { get; set; }
        public string MaxValue { get; set; }
        public Int32? Decimals { get; set; }
        public Boolean? PadDecimals { get; set; }
    }
}

File: Serenity.TypeScript.CoreLib/UI/Editors/DecimalEditor.ts
namespace Serenity {

    @Decorators.registerEditor('Serenity.DecimalEditor', [IDoubleValue])
    @Decorators.element('<input type="text"/>')
    export class DecimalEditor extends Widget<DecimalEditorOptions> implements IDoubleValue {

        constructor(input: JQuery, opt?: DecimalEditorOptions) {
            super(input, opt);

            input.addClass('decimalQ');
            var numericOptions = $.extend(Serenity.DecimalEditor.defaultAutoNumericOptions(), {
                vMin: Q.coalesce(this.options.minValue, '0.00'),
                vMax: Q.coalesce(this.options.maxValue, '999999999999.99')
            });

            if (this.options.decimals != null) {
                numericOptions.mDec = this.options.decimals;
            }

            if (this.options.padDecimals != null) {
                numericOptions.aPad = this.options.padDecimals;
            }

            (input as any).autoNumeric(numericOptions);
        }

        get_value(): number {
            var val = (this.element as any).autoNumeric('get');

            if (!!(val == null || val === ''))
                return null;

            return parseFloat(val);
        }

        get value(): number {
            return this.get_value();
        }

        set_value(value: number) {
            if (value == null || (value as any) === '') {
                this.element.val('');
            }
			else {
                (this.element as any).autoNumeric('set', value);
            }
        }

        set value(v: number) {
            this.set_value(v);
        }

        get_isValid(): boolean {
            return !isNaN(this.get_value());
        }

        static defaultAutoNumericOptions(): any {
            return {
                aDec: Q.Culture.decimalSeparator,
                altDec: ((Q.Culture.decimalSeparator === '.') ? ',' : '.'),
                aSep: ((Q.Culture.decimalSeparator === '.') ? ',' : '.'),
                aPad: true
            };
        }
    }

    export interface DecimalEditorOptions {
        minValue?: string;
        maxValue?: string;
        decimals?: any;
        padDecimals?: any;
    }

}